src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -13,7 +13,23 @@ namespace NuGet.PackageManagement.VisualStudio
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class VSSettings : Configuration.ISettings
     {
-        private Configuration.ISettings SolutionSettings { get; set; }
+        // to initialize SolutionSettings first time outside MEF constructor
+        private Configuration.ISettings _solutionSettings;
+
+        private Configuration.ISettings SolutionSettings
```

Comment:
Whether it's DTE Or not, the fact that you're moving `ResetSolutionSettings()` out of the ctor suggests there is a UI thread dependency, and as such, including a `ThrowIfNotOnUIThread()` method there is a good defense against deadlocks (exchanging them for a functional defect that is easy to diagnose, leading to a fast fix.